### PR TITLE
Listener class refactorings for java 8

### PR DESCRIPTION
**Pull request by Degubi**
Deprecate listener adapter classes, add default implementations for listener classes, add some missing overrides. Marked adapter classes as deprecated because they're not needed anymore.

### CODE REVIEW

File: src/java/org/jnativehook/keyboard/NativeKeyListener.java [hunk 1]
```diff
@@ -47,19 +47,19 @@
 	 *
 	 * @since 1.1
 	 */
-	public void nativeKeyTyped(NativeKeyEvent nativeEvent);
+	default void nativeKeyTyped(NativeKeyEvent nativeEvent) {}
```
Comment (kwhat) [inline src/java/org/jnativehook/keyboard/NativeKeyListener.java:50]: Sigh, they really screwed the backward compatibility pooch in Java 8.

**Comment by kwhat** [comment]
The PR is fine, however, I am concerned that the default keyword is going to screw backward compatibility for versions prior to 1.8.  Is there a way we can do want you are trying to do in a more compatible way?  I am looking to do multiple JRE releases moving forward, but I would still like to maintain some level of 1.5 compatibly.

**Comment by Degubi** [comment]
I really don't think so. If this isn't backwards compatible then there's no point to this request...  It can be closed then I think.

**Comment by kwhat** [comment]
I'll do some more digging and update accordingly.  Thank you.

**Comment by kwhat** [comment]
Java 8 & 11 is going into 2.2.

**Comment by kwhat** [comment]
@Degubi I am working on this right now, can you rebase against 2.2;  I think most of the issues are just that the files moved to src/main/java.

File: src/java/org/jnativehook/keyboard/NativeKeyListener.java (hunk 1 above)
Comment (kwhat) [reply src/java/org/jnativehook/keyboard/NativeKeyListener.java:50]: 1.5 support is is going away so this won't be a problem anymore.

**Comment by Degubi** [comment]
> @Degubi I am working on this right now, can you rebase against 2.2; I think most of the issues are just that the files moved to src/main/java.

Hi, yes I will do it when I get home.

**Comment by Degubi** [comment]
Reopening as a new pr because my original fork was deleted

**Pull request closed without merging**